modules/azure/disk.go

```diff
@@ -0,0 +1,103 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// DiskExists indicates whether the specified Azure Managed Disk exists
+// This function would fail the test if there is an error.
+func DiskExists(t testing.TestingT, diskName string, resGroupName string, subscriptionID string) bool {
+	exists, err := DiskExistsE(diskName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// DiskExistsE indicates whether the specified Azure Managed Disk exists
+func DiskExistsE(diskName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get the Disk object
+	_, err := GetDiskE(diskName, resGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetDiskType returns the Disk Storage Account Type of the Azure Managed Disk
+// This property also accessible from the VM client disk storage object but only works
```

Comment:
Your comment needs re-wording.  It's not clear which property on VirtualMachine you are referring to.  Instead of works, I would also reword to also accessible from <vm object path like virtualmachine.storageProfile.... or whatever that path is> and will error if the virtual machine is not running.  This call returns the disk type whether the VM is online or offline. or something to this effect.

Reply:
I removed the irrelevant parts - cc. @davesee to review (originally wrote the comment)

Reply:
Looks good to me. Made some changes on the VM PR as well since that uses the disk.go file.. will copy over these changes to my PR file.